crates/sui-framework/deps/move-stdlib/sources/type_name.move

```diff
@@ -24,6 +29,47 @@ module std::type_name {
         &self.name
     }
 
+    /// Get Address string (Base16 encoded), first part of the TypeName.
+    public fun get_address(self: &TypeName): String {
+        // Base16 (string) representation of an address has 2 symbols per byte.
+        let i = address::length() * 2;
+        let str_bytes = ascii::as_bytes(&self.name);
+        let addr_bytes = vector[];
+
+        // Reverse-read string from the last byte pushing the value
```

Comment:
I don't understand why we are reverse reading the address here?  IIUC, this means `addr_bytes` will be the reverse of the address prefix of `str_bytes`.

Reply:
yrros esrever ni gnikniht saw daeh ym